pkg/virt-controller/watch/vm_initializer.go

```diff
@@ -236,66 +236,77 @@ func checkPresetMergeConflicts(presetSpec *kubev1.DomainSpec, vmSpec *kubev1.Dom
 	return nil
 }
 
-func mergeDomainSpec(presetSpec *kubev1.DomainSpec, vmSpec *kubev1.DomainSpec) error {
-	err := checkPresetMergeConflicts(presetSpec, vmSpec)
-	if err != nil {
-		return err
-	}
+func mergeDomainSpec(presetSpec *kubev1.DomainSpec, vmSpec *kubev1.DomainSpec) (bool, error) {
+	presetConflicts := checkPresetMergeConflicts(presetSpec, vmSpec)
+	applied := false
+
 	if len(presetSpec.Resources.Requests) > 0 {
 		if vmSpec.Resources.Requests == nil {
 			vmSpec.Resources.Requests = k8sv1.ResourceList{}
-		}
-		for key, val := range presetSpec.Resources.Requests {
-			vmSpec.Resources.Requests[key] = val
+			for key, val := range presetSpec.Resources.Requests {
```

Comment:
Not a blocker, just a note:

This is stuff we need to think about.

If we have a list on teh VM, do we extend it or don't we touch it at all?

Reply:
`Requests` is actually a map of key-value pairs, not a list--so conflicting entries can easily be detected and accounted for in this case.